### PR TITLE
make: Updated Makefile.am to support distcheck

### PR DESCRIPTION
**Pull request by sshambar**
Updated Makefile.am to allow the make distcheck to work, which
means make install can be reversed by make uninstall, and
make clean correctly removes generated files.  make dist now
also creates a tarfile that can be used to configure and
build from.  Added README to install.

Signed-off-by: Scott Shambarger <devel@shambarger.net>

### CODE REVIEW

**Comment by commodo** [comment]
is this adding new build rules to `make`
if yes, can we add some CI calls to this in the [build.yml file](https://github.com/mstpd/mstpd/blob/master/.github/workflows/build.yml)

i can add them, if you tell me the make commands

**Comment by sshambar** [comment]
Well, it's not adding a target so much as fixing one :)  I added a CI entry for "make distcheck" and it passed on my fork.

I also added LICENSE to dist, and removed a couple generated files that shouldn't have been in the tar.

File: Makefile.am [hunk 1]
```diff
@@ -102,7 +128,7 @@ rpm: rpm.spec
 	git clone "$$SrcDir" "mstpd-$$Version" ; \
 	tar -cvjf "mstpd-$$Version.tar.bz2" "mstpd-$$Version" ; \
 	rm -rf "mstpd-$$Version" ; \
-	cd .. ; mkdir -p SPECS ; cp "$$SrcDir/rpm.spec" SPECS/mstpd.spec
+	cd .. ; mkdir -p SPECS ; cp "$$SrcDir/rpm.spec" SPECS/mstpd.spec; \
```
Comment (PaulSD) [inline Makefile.am:131]: This will cause a merge conflict with #119 ... You might want to skip it here to avoid that.

**Review by PaulSD — APPROVED** [review]
LGTM

File: Makefile.am (hunk 1 above)
Comment (sshambar) [reply Makefile.am]: Cool... was just fixing bug in the file (and perhaps 119 wouldn't be merged :), reverted the fix.

**Pull request merged**